libpromises/patches.c

```diff
@@ -226,15 +226,21 @@ int ExclusiveLockFile(int fd)
         .l_len = 0    /* till EOF */
     };
 
-    while (fcntl(fd, F_SETLKW, &lock) == -1)
+    if (wait)
     {
-        if (errno != EINTR)
+        while (fcntl(fd, F_SETLKW, &lock) == -1)
         {
-            return -1;
+            if (errno != EINTR)
+            {
+                return -1;
+            }
         }
+        return 0;
+    }
+    else
+    {
+        return fcntl(fd, F_SETLK, &lock);
```

Comment:
Hmm, on GNU/Linux man:fcntl(2) says:
```
RETURN VALUE
       For a successful call, the return value depends on the operation:

       F_DUPFD  The new file descriptor.

       F_GETFD  Value of file descriptor flags.

       F_GETFL  Value of file status flags.

       F_GETLEASE
                Type of lease held on file descriptor.

       F_GETOWN Value of file descriptor owner.

       F_GETSIG Value of signal sent when read or write becomes  possible,  or  zero  for  traditional
                SIGIO behavior.

       F_GETPIPE_SZ, F_SETPIPE_SZ
                The pipe capacity.

       F_GET_SEALS
                A bit mask identifying the seals that have been set for the inode referred to by fd.

       All other commands
                Zero.

       On error, -1 is returned, and errno is set appropriately.
```
And since we are not doing any of the explicitly listed operations, it also only returns `0` or `-1`. But better take care of this explicitly due to potential platform differences.